packages/zoltan2/src/algorithms/connectivity/Zoltan2_IceProp.hpp

```diff
@@ -0,0 +1,499 @@
+#include "Tpetra_Core.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Tpetra_Import.hpp"
+#include "Tpetra_FEMultiVector.hpp"
+#include "Tpetra_MultiVector_decl.hpp"
+#include "Tpetra_MultiVector_def.hpp"
+#include "Zoltan2_VtxLabel.hpp"
+
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <queue>
+#include <vector>
+
+#ifndef ZOLTAN2_ICEPROP_
+#define ZOLTAN2_ICEPROP_
+
+namespace Zoltan2{
+template<typename MAP>
```

Comment:
Why is this class templated on Map instead of local and global ordinal?

Reply:
Should I only template on local and global ordinal? Is it alright to fix the map type as a Tpetra Map?